keyboardClass.js

```diff
@@ -0,0 +1,241 @@
+                }
+            } else if (keyCode === 'ArrowLeft') {
+                event.preventDefault();
+                this.posIn = this.posIn > 0 ? this.posIn - 1 : 0;
+            } else if (keyCode === 'ArrowRight') {
+                event.preventDefault();
+                if (this.posIn < this.targetOfKeyboard.value.length) this.posIn += 1;
+            } else if (keyCode === 'ArrowUp') {
+                event.preventDefault();
+                this.upArrow();
+            } else if (keyCode === 'ArrowDown') {
+                event.preventDefault();
+                this.downArrow();
+            } else if ((keyCode === 'ShiftLeft' || keyCode === 'ShiftRight') && !this.shiftPressed && !this.btnList[keyCode].locked) {
+                this.shiftPressed = true;
+                this.changeCase();
+                if (this.btnList[keyCode].name === 'ShiftLeft') this.btnList.ShiftRight.locked = true;
+                else this.btnList.ShiftLeft.locked = true;
+            } else if (keyCode === 'ControlLeft' || keyCode === 'ControlRight') {
+                event.preventDefault();
+                this.ctrlPressed = true;
+            } else if (keyCode === 'AltLeft' || keyCode === 'AltRight') {
+                event.preventDefault();
+                this.altPressed = true;
+            } else if (this.btnList[keyCode].name === 'CapsLock') {
+                this.capsPressed = true;
+                this.changeCase();
+            }
+            if (this.ctrlPressed && this.altPressed) this.changeLanguage();
+            this.targetOfKeyboard.selectionStart = this.posIn;
+            this.targetOfKeyboard.selectionEnd = this.posIn;
+            if (!this.btnList[keyCode].locked) this.btnList[keyCode].addClass('active');
+        });
+        // shifr key mouse controls
+        this.domElement.addEventListener('mousedown', (event) => {
+            let keyCode = null;
+            if (event.target.tagName === 'BUTTON') keyCode = event.target.id;
+            else if (event.target.tagName === 'SPAN') keyCode = event.target.parentElement.id;
+            else return;
+
+            if (keyCode === 'ShiftLeft' || keyCode === 'ShiftRight') {
+                this.shiftPressed = true;
+                this.changeCase();
+                if (keyCode === 'ShiftLeft') this.btnList.ShiftRight.locked = false;
+                else this.btnList.ShiftLeft.locked = false;
+            } else if (keyCode === 'AltLeft' || keyCode === 'AltRight') {
+                this.altPressed = true;
+            } else if (keyCode === 'ControlLeft' || keyCode === 'ControlRight') {
+                this.ctrlPressed = true;
+            }
+        });
+        this.domElement.addEventListener('mouseup', (event) => {
+            let keyCode = null;
+            if (event.target.tagName === 'BUTTON') keyCode = event.target.id;
+            else if (event.target.tagName === 'SPAN') keyCode = event.target.parentElement.id;
+            else return;
```

Comment:
Please read about DRY and try to improve
https://ru.wikipedia.org/wiki/Don%E2%80%99t_repeat_yourself
https://refactoring.guru/smells/duplicate-code